lib/internal/Magento/Framework/Cache/Backend/RemoteSynchronizedCache.php

```diff
@@ -100,6 +100,10 @@ public function __construct(array $options = [])
         } elseif ($this->_options['remote_backend'] instanceof \Zend_Cache_Backend_ExtendedInterface) {
             $this->remote = $this->_options['remote_backend'];
         } else {
+            $dirPath = $this->_options['local_backend_options']['cache_dir'] ?? null;
```

Comment:
RemoteSynchronizedCache sholud not know about underline adapters. Please move this logic to adopters that works with  file system

Reply:
Another issue there - you're doing some logic in the object constructor, while this object might not be used at all. Can it be moved to some method and used in a proper place(s)?

Reply:
I will leave this PR as it is. Since it requires more refactoring. It uses direct calls to 
`\Zend_Cache::_makeBackend(` and checks library-related interfaces, so the proper place might be the `Cm_Cache_Backend_File` that is not part of the core code.

Reply:
Maybe in this case the right part will be the client code that creates this RemoteSynchronizedCache object?

Reply:
Not really. There is no such problem if the file cache was created natively. Looks like RC creates it a bit different way, so such problem exists. We could pass this config via callback in env.php but i don't think this is a good workaround. I expect that RC would create a file cache the same way magento does, and we should not play with directory checks in env.php
Anyways why should be responsible that the directory exists?

Reply:
In any case, it is not a good place for such code. Better to move it to https://github.com/colinmollenhour/Cm_Cache_Backend_File.
Also looks like if  you specify `hashed_directory_level`, the adapter will create a directory for you

Reply:
@ilnytskyi Yeah, seems like the correct fix will be delivering a fix to the https://github.com/colinmollenhour/Cm_Cache_Backend_File, so it will always create a directory if it doesn't exist?